src/Domain/QueryBuilders/BoolQuery.php

```diff
@@ -9,7 +9,7 @@
 use JeroenG\Explorer\Domain\Syntax\SyntaxInterface;
 use Webmozart\Assert\Assert;
 
-class BoolQuery implements QueryBuilderInterface
+class BoolQuery implements QueryBuilderInterface, SyntaxInterface
```

Comment:
What is the benefit of having the SyntaxInterface here as well?

Reply:
You can use bool queries inside bool queries.
Example:
```      $search = Dive::search();

        if ($findDivesCommand->getKeywords()) {
            $query = new BoolQuery();
            $query->should(new Nested('place', new Matching('place.name', $findDivesCommand->getKeywords())));
            $query->should(new Nested('buddies', new Matching('buddies.name', $findDivesCommand->getKeywords())));
            $query->should(new Nested('tags', new Matching('tags.text', $findDivesCommand->getKeywords())));
            $search->must($query);
      }
``` 
See [this](https://github.com/blackshadev/littledivelogserver/pull/23/files#diff-f4dda2c421d1915bfa8b8592fb201a7ed6ccc0eeb602384d51b2d4e7fa535983R171) diff where I used it.

Reply:
Cool, thanks for the diff!